docs/architecture/mxgraph-integration.adoc

```diff
@@ -0,0 +1,89 @@
+
+On `mxGraph` graph creation, the lib fills the following group of BPMN elements:
+
+* `pool` is the parent of `lanes` and `inter-lane elements` (for instance, `message flows`)
+* `lane` is the parent of elements it includes in the BPMN source
+
+The default mxGraph parent is the parent of
+
+* inter pool elements (for instance, `message flows`)
+* elements when the BPMN source doesn't define any lane/pool
+
+
+==== BPMN elements positioning
+
+*Note*: the following applies to both BPMN shapes and edges (including waypoints)
+
+The bpmn input source coordinates are absolute whereas mxgraph uses coordinates in the referential of the parent cell.
+As we define a parent-child relationship when inserting BPMN elements in the mxGraph model, a coordinate transformation
+layer is required. See `MxGraphRenderer` for more details.
+
+
+===== Resources
+
+====== mxGraph
+
+.https://jgraph.github.io/mxgraph/docs/manual.html#3.1.3.2[mxgraph Geometry]
+[quote]
+The coordinate system in JavaScript is x is positive to the right and y is positive
+downwards, and in terms of the graph, the positioning is absolute to the container
+within which the mxGraph is placed.
+
+.https://jgraph.github.io/mxgraph/docs/manual.html#3.1.4[mxgraph Group Structure]
+[quote]
+The x,y position of a vertex is its position relative to its parent, so in the case of
+default grouping (all cells sharing the default parent) the cell positioning is also
+the absolute co-ordinates on the graph component.
+
+
+====== BPMN specification
+
+.BPMNPlane (page 371)
+[quote]
+BPMNPlane element is always owned by a BPMNDiagram and represents the root diagram element of that diagram.
+The plane represents a 2 dimensional surface with an origin at (0, 0) along the x and y axes with increasing coordinates
+to the right and bottom. Only positive coordinates are allowed for diagram elements that are nested in a BPMNPlane.
+
+.BPMNShape (page 372)
+[quote]
+All BPMNShape elements are owned directly by a BPMNPlane (that is the root element in a BPMNDiagram), i.e., shapes
+are not nested within each other in the BPMN DI model although they may appear that way when depicted. The bounds
+of a BPMNShape are always relative to that plane’s origin point and are REQUIRED to be positive coordinates. Note that
+the bounds’ x and y coordinates are the position of the upper left corner of the shape (relative to the upper left corner of
+the plane).
+
+
+.BPMNEdge (page 375)
+[quote]
+All BPMNEdge elements are owned directly by a BPMNPlane (that is the root element in a BPMNDiagram). The
+waypoints of BPMNEdge are always relative to that plane’s origin point and are REQUIRED to be positive coordinates.
+
+.BPMNLabel (page 377)
+[quote]
+The bounds of BPMNLabel are always relative to the containing plane’s origin point. Note that the bounds’ x and y
+coordinates are the position of the upper left corner of the label (relative to the upper left corner of the plane).
```

Comment:
```suggestion
The bounds of BPMNLabel are always relative to the containing plane’s origin point. Note that the x and y coordinates of the bounds are the position of the upper left corner of the label (relative to the upper left corner of the plane).
```

Reply:
it is the strict copy of the BPMN specification content (same remark as above)